### PR TITLE
[FIXED] Deal with misaligned DeleteBlocks in SyncDeleted

### PR DESCRIPTION
**Pull request by sciascid**
SyncDeleted only skipped delete blocks when the incoming and local DeleteBlocks matched at the exact same index. This stopped working as soon as the incoming has shifted or new delete blocks appear in the list.
This commit adds method pruneDeleteBlock which lift the above limitation. It takes an incoming DeleteBlock and a list of local DeleteBlocks. If pruneDeleteBlock finds that the local DeleteBlocks has a matching entry for the incoming DeleteBlock, then it will return true indicating that it can be skipped, avoiding unnecessary deletes.

Signed-off-by: Daniele Sciascia <daniele@nats.io>

### CODE REVIEW

**Comment by sciascid** [comment]
@claude @codex review

File: server/filestore.go [hunk 1]
```diff
@@ -12119,6 +12111,34 @@ func (fs *fileStore) SyncDeleted(dbs DeleteBlocks) error {
...
+	// Drop blocks that end before db starts.
+	for bLast < aFirst {
+		blocks = blocks[1:]
+		if len(blocks) == 0 {
+			return false, blocks
+		}
+		bFirst, bLast, bNum = blocks[0].State()
+	}
+
+	if aFirst == bFirst && aLast == bLast && aNum == bNum {
```
[the remark this answers is not in the document]
Comment (sciascid) [reply server/filestore.go:12135]: @codex I also had some doubt about the condition for matching. However, I think the correctness follows from the fact that sequence numbers are created / deleted at most once. For example: if a local sequenceset represents deleted sequences {1, 5,10}, the corresponding DeleteBlock would have State() first:1 last:10  num: 3. If this node then receives a snapshot containing  DeleteBlock with the same 1,10,3 state, but the underlying deletes were {1, 4, 10}... then how would you explain that deleted sequence 5 is no longer deleted?  If something like that happens the it must be that the state had already diverged in some way? And if so, that would be a bug somewhere else...
Let me know if I'm missing something else.

[the remark this answers is not in the document]
Comment (sciascid) [reply server/filestore.go:12135]: Also, notice that this condition for matching DeleteBlock has not changed in this commit. This specific part should not introduce regressions.

**Comment by sciascid** [comment]
@claude the test has been extended to cover the case you mentioned.
About the potential edge case with overlapping ranges: overlapping ranges are not handled, and are expected to result in unnecessary deletes. However, removing a message appears to be idempotent. Don't think there are any risks with "double-deletes" except for unnecessary work.
